src/components/forms/ComboBox/ComboBox.tsx

```diff
@@ -0,0 +1,317 @@
+    }
+  }
+
+  const handleInputClick = (): void => {
+    setFocusMode(FocusMode.Input)
+    setIsOpen(true)
+    setFocusedValue(filteredOptions[0].id)
+  }
+
+  const handleInputBlur = (event: FocusEvent<HTMLInputElement>): void => {
+    const target = event.relatedTarget
+
+    if (
+      !target ||
+      (target instanceof Node && !containerRef.current?.contains(event.target))
+    ) {
+      setFocusMode(FocusMode.None)
+      setIsOpen(false)
+    }
+  }
+
+  const handleListItemKeyDown = (event: KeyboardEvent): void => {
+    if (event.key == 'Escape') {
+      setIsOpen(false)
+      if (!selectedValue) {
+        setInputValue('')
+      }
+      setFocusedValue(undefined)
+      setFocusMode(FocusMode.Input)
+    } else if (event.key == 'ArrowDown' || event.key == 'Down') {
+      event.preventDefault()
+      changeFocus(Direction.Down)
+    } else if (event.key == 'ArrowUp' || event.key == 'Up') {
+      event.preventDefault()
+      changeFocus(Direction.Up)
+    }
+  }
+
+  const handleListItemClick = (option: ComboBoxOption): void => {
+    selectValue(option.id)
+    setInputValue(option.label || option.id)
+    setIsOpen(false)
+    setFocusMode(FocusMode.Input)
+  }
+
+  const handleListItemMouseMove = (option: ComboBoxOption): void => {
+    if (focusedValue !== option.id) {
+      setFocusedValue(option.id)
+      setFocusMode(FocusMode.Item)
+    }
+  }
+
+  const handleArrowClick = (): void => {
+    if (isOpen) {
+      setFocusMode(FocusMode.Input)
+      setIsOpen(false)
+    } else {
+      if (selectedValue) {
+        setFocusMode(FocusMode.Item)
+      }
+      setIsOpen(true)
+    }
+  }
+
+  const handleClearClick = (): void => {
+    setInputValue('')
+    setSelectedValue(undefined)
+    setFocusMode(FocusMode.Input)
+  }
+
+  const containerClasses = classnames('usa-combo-box', className, {
+    'usa-combo-box--pristine': selectedValue,
+  })
+
+  return (
+    <div
+      data-testid="combo-box"
+      className={containerClasses}
+      id={id}
+      ref={containerRef}>
+      <select
+        className="usa-select usa-sr-only usa-combo-box__select"
+        name={name}
+        aria-hidden
+        tabIndex={-1}
+        value={selectedValue}
+        {...selectProps}>
+        {options.map((option) => (
+          <option key={option.id} value={option.id}>
+            {option.label || option.id}
+          </option>
+        ))}
+      </select>
+      <input
+        data-testid="combo-box-input"
+        aria-owns={listID}
+        aria-autocomplete="list"
+        aria-describedby={assistiveHintID}
+        aria-expanded={isOpen}
+        aria-controls=""
+        id={selectID}
+        className="usa-combo-box__input"
+        type="text"
+        role="combobox"
+        ref={inputRef}
+        value={inputValue}
+        onBlur={handleInputBlur}
+        onClick={handleInputClick}
+        onKeyDown={handleInputKeyDown}
+        onChange={(e): void => setInputValue(e.target.value)}
```

Comment:
_*I think*_ we want to allow the consumer to still be able to call their own `onChange` passed in from the `selectProps.onChange`